Tests/SonarQube.TeamBuild.PreProcessor.Tests/SonarWebServiceTest.cs

```diff
@@ -25,20 +25,34 @@ public class SonarWebServiceTest
         public void Init()
         {
             downloader = new TestDownloader();
-            ws = new SonarWebService(downloader, "http://localhost:9000", "cs", "fxcop");
+            ws = new SonarWebService(downloader, "http://localhost:9000");
         }
 
         [TestMethod]
-        public void GetQualityProfile()
+        public void TryGetQualityProfile()
         {
+            bool result;
+            string qualityProfile;
+
             downloader.Pages["http://localhost:9000/api/profiles/list?language=cs&project=foo%20bar"] = "[{\"name\":\"profile1\",\"language\":\"cs\",\"default\":true}]";
-            Assert.AreEqual("profile1", ws.GetQualityProfile("foo bar"));
+            result = ws.TryGetQualityProfile("foo bar", "cs", out qualityProfile);
+            Assert.IsTrue(result);
+            Assert.AreEqual("profile1", qualityProfile);
 
-            downloader.Pages["http://localhost:9000/api/profiles/list?language=cs"] = "[{\"name\":\"profile1\",\"language\":\"cs\",\"default\":true},{\"name\":\"profile2\",\"language\":\"cs\",\"default\":false}]";
-            Assert.AreEqual("profile1", ws.GetQualityProfile("bar"));
+            downloader.Pages["http://localhost:9000/api/profiles/list?language=vbnet"] = "[{\"name\":\"profile1\",\"language\":\"vbnet\",\"default\":true},{\"name\":\"profile2\",\"language\":\"vbnet\",\"default\":false}]";
```

Comment:
Coverage would be better if you'd setup the download with all the keys and then request them